packages/orchestration/src/exos/chain-hub.js

```diff
@@ -425,6 +438,18 @@ export const makeChainHub = (zone, agoricNames, vowTools) => {
       }
       return undefined;
     },
+    /**
+     * @param {string} address bech32 address
+     * @returns {CosmosChainInfo}
+     */
+    getChainInfoByAddress(address) {
+      const prefix = getBech32Prefix(address);
+      if (bech32PrefixToChainName.has(prefix)) {
+        const chainName = bech32PrefixToChainName.get(prefix);
+        return chainInfos.get(chainName);
+      }
+      throw makeError(`Chain info not found for bech32Prefix ${q(prefix)}`);
```

Comment:
consider: generally better to validate before getting to the happy path:
```suggestion
      if (!bech32PrefixToChainName.has(prefix)) {
        throw makeError(`Chain info not found for bech32Prefix ${q(prefix)}`);
      }

      const chainName = bech32PrefixToChainName.get(prefix);
      return chainInfos.get(chainName);
```